### PR TITLE
when looking for latest fastly VCL to clone, use only active VCLs

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves:

Resolves https://github.com/LLK/scratch-www/issues/1342

### Changes:

When comparing fastly vcls to find most recent one, prefer one vcl to another if and only if it is both higher-numbered and active. (Previously, we only considered number.)

### Test Coverage:

Added a test.

### CODE REVIEW

File: bin/lib/fastly-extended.js [hunk 1]
```diff
@@ -39,7 +39,14 @@ module.exports = function (apiKey, serviceId) {
             }
             var latestVersion = versions.reduce(function (lateVersion, version) {
```
Comment (colbygk) [inline bin/lib/fastly-extended.js]: I'm surprised our linter didn't warn/error on this. Don't we prefer (lateVersion, version) => { ... } ? Also, the var statements here ought to be const's. But that is pre-existing, so a new issue and not this PR.

File: test/unit/lib/fastly-extended.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                    active: false
+                },
+                {
+                    number: 1,
+                    active: true
+                },
+                {
+                    number: 2,
+                    active: true
+                },
```
Comment (colbygk) [inline test/unit/lib/fastly-extended.test.js]: I don think this is good test data. Two versions cannot be active at the same time for a single service by definition no?

File: bin/lib/fastly-extended.js (hunk 1 above)
Comment (colbygk) [reply bin/lib/fastly-extended.js]: Also to avoid confusion, I think the two objects being compared in the reduce() function ought not be named the same as the object being returned via callback at line 52 and defined/getting return from the reduce() at line 40.

Comment (colbygk) [reply bin/lib/fastly-extended.js]: (sorry, just noticed they're not the same name, just close named)

File: test/unit/lib/fastly-extended.test.js (hunk 2 above)
Comment (benjiwheeler) [reply test/unit/lib/fastly-extended.test.js]: Good point, changing!

**Comment by benjiwheeler** [comment]
OK, I revised and (I think) simplified the `latestActiveVersionSoFar()` code, made the tests only include one active version, and added more tests for edge cases. I also changed `latestActiveVersionSoFar()` so that if there is no active version, it returns null.
